CatProducer/plugins/CATElectronProducer.cc

```diff
@@ -165,11 +240,20 @@ cat::CATElectronProducer::produce(edm::Event & iEvent, const edm::EventSetup & i
     double phIso04 = aElectron.photonIso(0.4);
     aElectron.setrelIso(0.4, chIso04, nhIso04, phIso04, elEffArea04, rhoIso, ecalpt);
 
+
+    //////////////// pfcands  //////////////////                                                                                                                                                                                                                       
+    edm::Handle<pat::PackedCandidateCollection> pfcands;
+    iEvent.getByToken(pfSrc_, pfcands);
+
+
+
     double elEffArea03 = getEffArea( 0.3, scEta);
     double chIso03 = aElectron.chargedHadronIso(0.3);
     double nhIso03 = aElectron.neutralHadronIso(0.3);
     double phIso03 = aElectron.photonIso(0.3);
     aElectron.setrelIso(0.3, chIso03, nhIso03, phIso03, elEffArea03, rhoIso, ecalpt);
+    aElectron.setrelMiniIso(getMiniRelIso( pfcands, dynamic_cast<const reco::Candidate *>(&aElectron), 0.05, 0.2, 10., rhoIso,elEffArea03));
+
```

Comment:
I simply copied the function from some code I was given. I removed the dynamic cast. I will make a new pull request.